lib/improver/nbhood/nbhood.py

```diff
@@ -147,6 +147,55 @@ def adjust_nsize_for_ens(ens_factor, num_ens, width):
                          math.sqrt((width**2.0)/num_ens))
         return new_width
 
+    @staticmethod
+    def check_cube_coordinates(cube, new_cube):
+        """
+        Find and promote to dimension coordinates any scalar coordinates in
+        new_cube that were originally dimension coordinates in the progenitor
+        cube.
+
+        Parameters
+        ----------
+        cube : iris.cube.Cube
+            The input cube provided to nbhood.
+        new_cube : iris.cube.Cube
+            The cube produced by the neighbourhooding process that must be
+            checked for demoted dimensional coordinates.
+
+        Returns
+        -------
+        new_cubewidth : iris.cube.Cube
+            Modified neighbourhooded cube with relevant scalar coordinates
+            promoted to dimension coordinates.
+
+        Raises
+        ------
+        iris.exceptions.CoordinateNotFoundError raised if the final dimension
+        coordinates of the returned cube do not match the input cube.
+
+        """
+        # Promote available and relevant scalar coordinates
+        for coord in new_cube.aux_coords[::-1]:
+            if coord in cube.dim_coords:
+                new_cube = iris.util.new_axis(new_cube, coord)
+
+        # Ensure dimension order matches; if lengths are unequal a coordinate
```

Comment:
Looking at the acceptance tests input and output you are losing the bounds on your dimension coordinate through the processing chain. I suggest we have a look. I've put an ipython notebook link on JIRA.